contrib/babelfishpg_tsql/sql/sys_functions.sql

```diff
@@ -3297,3 +3297,10 @@ END;
 $BODY$
 LANGUAGE plpgsql;
 GRANT EXECUTE ON FUNCTION sys.INDEXPROPERTY(IN object_id INT, IN index_or_statistics_name sys.nvarchar(128),  IN property sys.varchar(128)) TO PUBLIC;
+
+CREATE OR REPLACE FUNCTION sys.APP_NAME() RETURNS SYS.NVARCHAR(128)
+AS
+$$
+    SELECT current_setting('application_name');
+$$
+LANGUAGE sql PARALLEL SAFE IMMUTABLE RETURNS NULL ON NULL INPUT;
```

Comment:
I feel like `IMMUTABLE` is not correct. Please refer https://www.postgresql.org/docs/current/xfunc-volatility.html

Reply:
In SQL Server, application name can't be changed in a session. However, it can change across session. If you declare the function as IMMUTABLE, it'll cache the value in some cases and produce incorrect result. In fact, it can persist the value in some cases as well.

For T-SQL functions, the easiest way to check whether a function can be immutable or not is by creating a persisted computed column using the function in the expression.

Consider the following scenario in SQL Server:
```
1> SET QUOTED_IDENTIFIER ON
2> GO
1> CREATE TABLE t1(a int, b as APP_NAME() PERSISTED)
2> GO
Msg 4936, Level 16, State 1, Server EC2AMAZ-R3QO7RM, Line 1
Computed column 'b' in table 't1' cannot be persisted because the column is non-deterministic.
```

It doesn't allow persisted column with APP_NAME() because the value will change across different connections. So, it will not be immutable or deterministic and the output differs based on the state of the database. Hence, in this case the function should be declared as STABLE.

Reply:
> Yes, it should be STABLE. In SQL Server, application name can't be changed in a session. However, it can change across session. If you declare the function as IMMUTABLE, it'll cache the value in some cases and produce incorrect result. In fact, it can persist the value in some cases as well.
> 
> For T-SQL functions, the easiest way to check whether a function can be immutable or not is by creating a persisted computed column using the function in the expression.
> 
> Consider the following scenario in SQL Server:
> 
> ```
> 1> SET QUOTED_IDENTIFIER ON
> 2> GO
> 1> CREATE TABLE t1(a int, b as APP_NAME() PERSISTED)
> 2> GO
> Msg 4936, Level 16, State 1, Server EC2AMAZ-R3QO7RM, Line 1
> Computed column 'b' in table 't1' cannot be persisted because the column is non-deterministic.
> ```
> 
> It doesn't allow persisted column with APP_NAME() because the value will change across different connections. So, it will not be immutable or deterministic and the output differs based on the state of the database. Hence, in this case the function should be declared as STABLE.

That makes a lot of sense. Thanks for the clarification!